js/compiler/compiler.go

```diff
@@ -148,34 +165,42 @@ type babel struct {
 	vm        *goja.Runtime
 	this      goja.Value
 	transform goja.Callable
-	mutex     sync.Mutex // TODO: cache goja.CompileAST() in an init() function?
+	m         sync.Mutex
 }
 
 func newBabel() (*babel, error) {
 	var err error
 
-	once.Do(func() {
-		vm := goja.New()
-		if _, err = vm.RunString(babelSrc); err != nil {
-			return
-		}
-
-		this := vm.Get("Babel")
-		bObj := this.ToObject(vm)
-		globalBabel = &babel{vm: vm, this: this}
-		if err = vm.ExportTo(bObj.Get("transform"), &globalBabel.transform); err != nil {
+	onceBabelCode.Do(func() {
+		globalBabelCode, err = goja.Compile("<eval>", babelSrc, false)
+		if err != nil { // TODO cache this?
 			return
 		}
```

Comment:
Why would we need to cache it? Actually, is the `err` check and return here needed at all?

Reply:
in practice ... it should never be an error, I am almost fine with panicking as if babel doesn't compile ... well there is nothing we can do anymore ;)

Reply:
I'm not sure you answered my question :sweat_smile: If Babel doesn't compile then this would only fail once, so what's the point of caching it and of the `TODO`? And since we check this `err` a few lines below this, I don't see the point of this additional check and `return` from this lambda.

Reply:
OKay .. the whole `if` is ... kind of not needed as this whole thing is inside `once.Do`.

But the idea behind the caching is that k6 will execute this `once.Do` .. once and only the first call will actually set the `err` for it's execution, all other executions will block on the `once.Do` until finished and will return a `nil` `globalBabelCode`

Reply:
Ah, I see, so as a way to prevent blocking other goroutines? I would at least explain that in the `TODO`, as it's unclear why right now. Though given this is such an unlikely edge case I would just remove the check and `TODO`.